src/cdk/overlay/overlay-ref.ts

```diff
@@ -488,34 +472,49 @@ export class OverlayRef implements PortalOutlet {
     }
   }
 
-  /** Detaches the overlay content next time the zone stabilizes. */
+  /** Detaches the overlay once the content finishes animating and is removed from the DOM. */
   private _detachContentWhenEmpty() {
-    // Normally we wouldn't have to explicitly run this outside the `NgZone`, however
-    // if the consumer is using `zone-patch-rxjs`, the `Subscription.unsubscribe` call will
-    // be patched to run inside the zone, which will throw us into an infinite loop.
-    this._ngZone.runOutsideAngular(() => {
-      // We can't remove the host here immediately, because the overlay pane's content
-      // might still be animating. This stream helps us avoid interrupting the animation
-      // by waiting for the pane to become empty.
-      const subscription = this._renders
-        .pipe(takeUntil(merge(this._attachments, this._detachments)))
-        .subscribe(() => {
-          // Needs a couple of checks for the pane and host, because
-          // they may have been removed by the time the zone stabilizes.
-          if (!this._pane || !this._host || this._pane.children.length === 0) {
-            if (this._pane && this._config.panelClass) {
-              this._toggleClasses(this._pane, this._config.panelClass, false);
-            }
-
-            if (this._host && this._host.parentElement) {
-              this._previousHostParent = this._host.parentElement;
-              this._host.remove();
-            }
-
-            subscription.unsubscribe();
-          }
-        });
-    });
+    // Attempt to detach on the next render.
+    try {
+      this._detachContentAfterRenderRef = afterNextRender(() => this._detachContent(), {
```

Comment:
Do we also want to run `_detachContent` once initially in case the content is already gone?

Reply:
the `afterNextRender` should catch that case as well, if its gone now it'll be gone on the next render as well